internal/telemetry/tracker_test.go

```diff
@@ -0,0 +1,103 @@
+package telemetry
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+type testClient struct {
+	calls []interface{}
+}
+
+func (client *testClient) Enqueue(track analytics.Message) error {
+	client.calls = append(client.calls, track)
+	return nil
+}
+
+func (client *testClient) Close() error {
+	return nil
+}
+
+const (
+	testID          = "id123"
+	testUserID      = "user123"
+	testExecutionID = "execution123"
+)
+
+var testTime = time.Date(2021, 1, 2, 3, 4, 5, 6, time.UTC)
+
+func TestNoopTracker(t *testing.T) {
+	t.Run("should create a noop tracker and should not do anything when track is called", func(t *testing.T) {
+		tracker := noopTracker{}
+		testEvent := createEvent(EventTypeCommandStart, nil, "someCommand")
+		testTrackStdoutOutput(t, &tracker, testEvent, "")
+	})
+}
+
+func TestStdoutTracker(t *testing.T) {
+	t.Run("should create an stdout tracker and should print the tracking information to stdout", func(t *testing.T) {
+		tracker := stdoutTracker{}
+		testTrackStdoutOutput(
+			t,
+			&tracker,
+			createEvent(EventTypeCommandComplete, nil, "someCommand"),
+			"03:04:05 UTC TELEM someCommand: COMMAND_COMPLETE[]\n",
+		)
+	})
+}
+
+func TestSegmentTracker(t *testing.T) {
+	client := &testClient{}
+	t.Run("should create the segment tracker and should print the tracking information to the logger", func(t *testing.T) {
+		tracker := segmentTracker{}
+		tracker.client = client
+		tracker.Track(createEvent(EventTypeCommandError, []EventData{{Key: EventDataKeyErr, Value: "Something"}}, "someCommand"))
+		testClientResults := tracker.client.(*testClient)
+		assert.Equal(t, 1, len(testClientResults.calls))
+
+		expectedTrack := analytics.Track{
+			MessageId: testID,
+			UserId:    testUserID,
+			Timestamp: testTime,
+			Event:     string(EventTypeCommandError),
+			Properties: map[string]interface{}{
+				string(EventDataKeyErr): "Something",
+			},
+		}
+		actualTrack := testClientResults.calls[0].(analytics.Track)
+		assert.NotNil(t, actualTrack)
+		assert.Equal(t, 1, len(actualTrack.Properties))
+		assert.Equal(t, expectedTrack, actualTrack)
+	})
+}
+
+func testTrackStdoutOutput(t *testing.T, tracker Tracker, event event, expected string) {
```

Comment:
left you a comment about moving this code from a helper to the test case itself, but here are a few other comments to clean up:
* always check the error (you're ignoring errors from `os.Pipe`, `w.Close()`, & `ioutil.ReadAll` here), you have `t *testing.T` here so you can `assert.Nil` easily
* in general, when using helper functions that pass around `t *testing.T` (presumably so you can make `assert` calls), you'll want to make sure to call `t.Helper()` at the top/first thing of the helper function (it ensures proper test output if/when a test fails)
  * fwiw, if/when you move this code to the test func itself, `t.Helper()` will become unnecessary
* b/c you have the defer which replaces `os.Stdout` with the original reference of it, you won't need to do it again on L#89 here